tests_e2e/tests/agent_removal/agent_removal.py

```diff
@@ -0,0 +1,118 @@
+        elif not hasattr(context, "expected_versions") and not hasattr(context, "removed_version"):
+            fail("Exactly one of the following arguments should be provided: expected_versions, removed_version")
+        elif hasattr(context, "expected_versions"):
+            self._expected_versions = context.expected_versions.split(';')
+        elif hasattr(context, "removed_version"):
+            self._removed_version = context.removed_version
+        self._GAFamily = context.GAFamily if hasattr(context, "GAFamily") else "Prod"
+
+    def run(self):
+        log.info("Retrieving agent manifests for {0} GAFamily...".format(self._GAFamily))
+
+        manifest_uris = self._get_manifest_uris()
+
+        log.info("")
+        log.info("Checking agent versions in manifest URIs from goal state...")
+        for uri in manifest_uris:
+            log.info("")
+            log.info("URI: {0}".format(uri))
+            agent_versions = self._ssh_client.run_command("curl -s {0} | grep -oP '(?<=<Version>).*?(?=</Version>)'".format(uri)).splitlines()
+            log.info("Agent versions in manifest: {0}".format(agent_versions))
+            if self._expected_versions is not None:
+                log.info("Expected versions: {0}".format(self._expected_versions))
+                assert_that(self._expected_versions).described_as("Expected agent versions in manifest does not match actual agent versions in manifest. Expected={0}, Actual={1}".format(self._expected_versions, agent_versions)).is_equal_to(agent_versions)
+                log.info("Expected versions match actual agent versions in manifest")
+            else:
+                log.info("Agent version which was deleted: {0}".format(self._removed_version))
+                assert_that(agent_versions).described_as("Removed version {0} is still in manifest. Manifest versions: {1}".format(self._removed_version, agent_versions)).does_not_contain(self._removed_version)
+                log.info("Agent version which was deleted is not in manifest")
+
+        log.info("")
+        log.info("Validated all manifests successfully.")
+
+    def _get_manifest_uris(self):    #pylint: disable=R1710 (inconsistent-return-statements)
+        # Try to get manifest URIs from current goal state, allowing several retries in case current goal state hasn't
+        # been written to history folder yet.
+        manifest_uris = []
+        for _ in range(4):
+            current_gs_history_path = self._ssh_client.run_command("find /var/lib/waagent/history -mindepth 1 -type d", use_sudo=True).rstrip()
+
+            # Try to get extensions config for current goal state
+            ext_config_path = os.path.join(current_gs_history_path, _EXT_CONF_FILE_NAME)
+            try:
+                ext_config_text = self._ssh_client.run_command("cat {0}".format(ext_config_path), use_sudo=True)
+                xml_doc = parse_doc(ext_config_text)
```

Comment:
We have all this logic in agent, can we import agent module instead of duplicate similar logic here? If we need to change logic around this area then we need it here as well

We have done similarly in other tests. Reference: https://github.com/Azure/WALinuxAgent/blob/master/tests_e2e/tests/scripts/agent_update-wait_for_rsm_gs.py#L32

Reply:
Updated